metricbeat/module/windows/perfmon/pdh_windows.go

```diff
@@ -95,11 +102,11 @@ func NewQuery(dataSource string) (*Query, error) {
 
 	return &Query{
 		handle:   h,
-		counters: map[string]PdhCounterHandle{},
+		counters: make(Counters),
 	}, nil
 }
 
-func (q *Query) AddCounter(counterPath string) error {
+func (q *Query) AddCounter(counterPath string, format string) error {
```

Comment:
From an API perspective I would not allow a free-form string for `format`. I would declare an [enum](https://golang.org/doc/effective_go.html#constants) for the allowed format types. And then inside of the function reject unknown `format` enum values with an error.